packages/react-sdk/src/components/image.tsx

```diff
@@ -1,24 +1,187 @@
+    }
+
+    if (percentSizes.length) {
+      // we can exclude from srcSets all images which are smaller than the smallestRatio * smallesDeviceSize
+      const smallestRatio = Math.min(...percentSizes) * 0.01;
+      return {
+        widths: allSizes.filter((s) => s >= deviceSizes[0] * smallestRatio),
+        kind: "w",
+      };
+    }
+    return { widths: allSizes, kind: "w" };
+  }
+
+  if (width == null) {
+    return { widths: deviceSizes, kind: "w" };
+  }
+
+  const widths = [
+    ...new Set(
+      [width, width * 2].map(
+        (w) => allSizes.find((p) => p >= w) || allSizes[allSizes.length - 1]
+      )
+    ),
+  ];
+  return { widths, kind: "x" };
+};
+
+const generateImgAttrs = ({
+  src,
+  width,
+  quality,
+  sizes,
+}: {
+  src: string;
+  quality: number;
+  width: number | undefined;
+  sizes: string | undefined;
+}): {
+  src: string;
+  srcSet: string | undefined;
+  sizes: string | undefined;
+} => {
+  const { widths, kind } = getWidths(width, sizes);
+
+  return {
+    sizes: !sizes && kind === "w" ? "100vw" : sizes,
+    srcSet: widths
+      .map(
+        (w, i) =>
+          `${cloudflareImageLoader({ src, quality, width: w })} ${
+            kind === "w" ? w : i + 1
+          }${kind}`
+      )
+      .join(", "),
+
+    // Must be last, to prevent Safari to load images twice
+    src: cloudflareImageLoader({
+      src,
+      quality,
+      width: widths[widths.length - 1],
+    }),
+  };
+};
+
+const getInt = (x: unknown): number | undefined => {
+  if (typeof x === "number") {
+    return Math.round(x);
+  }
+
+  if (typeof x === "string" && /^[0-9]+$/.test(x)) {
+    return parseInt(x, 10);
+  }
+
+  return undefined;
+};
+
+const DEFAULT_QUALITY = 80;
+
+/**
+ * DEFAULT_SIZES Just an assumption that most images (except hero and icons) are 100% wide on mobile and 50% on desktop.
+ * For icons width are usually set explicitly so DEFAULT_SIZES is not applied.
+ * For hero images, we can allow in UI to select sizes=100vw explicitly.
+ * Anyway, the best would be to calculate this based on canvas data from different breakpoints.
+ **/
+const DEFAULT_SIZES = "(min-width: 1280px) 50vw, 100vw";
 
 export const Image = forwardRef<ElementRef<typeof defaultTag>, ImageProps>(
-  (props, ref) => <img {...props} ref={ref} />
+  (props, ref) => {
+    const width = getInt(props.width);
+    const quality = Math.max(
+      Math.min(getInt(props.quality) ?? DEFAULT_QUALITY, 100),
+      0
+    );
+
+    const sizes = props.sizes ?? (width == null ? DEFAULT_SIZES : undefined);
+
+    const imageAttributes =
+      props.src != null && props.src != ""
+        ? generateImgAttrs({
+            src: props.src,
+            width,
+            quality,
+            sizes,
+          })
+        : { src: imagePlaceholderSvg };
+
+    return <img {...props} {...imageAttributes} decoding="async" ref={ref} />;
+  }
 );
 
 Image.defaultProps = {
   src: "",
   loading: "lazy",
-  width: "auto",
```

Comment:
Removed width/height from UI. Need to be returned with "" and transformed through getInt() inside `image`

Reply:
Default props have been a major pain in that ass also because we need a way to say what props we want to __always__ show for a component, because they are frequently used and the rest can be added by the user.

So we def. need to decouple defaultProps.

Reply:
So I am open to proposals on how we can do both, what you need and expressing props shown by default on props panel

Reply:
Btw. this is the issue I was talking about, wanna take it next?

Reply:
https://github.com/webstudio-is/webstudio-designer/issues/192